README.md

```diff
@@ -170,155 +170,50 @@
-    ```typescript
-    {
-      clientIdentifier: string;
-      clientSecret: string;
-    }
-    ```
-
-  - Response payload
-
-    type: object
-    ```typescript
-    {
-      authToken: string;
-      cloudHost: string;
-      account: string;
-      accountId: string;
-      company: string;
-      companyId: string;
-      selectedLocale: string;
-      user: string;
-      userId: string;
-      initialContext: [
-        {
-          name: string;
-          type: string;
-          value: any;
-        }
-      ];
-      screenConfiguration: Object | null;
-    }
-    ```
-
-- #### V1.FLOWS.CAN_CONTINUE  
-  Used to indicate if current flow step ready to continue to the next step. Shell host will
-  block switching to the next step until this event sent with true value in payload. If data
-  at some point becames invalid again, this event with false value in payload can be sent
-  in order to block switching to next step again.
-
-  - Request payload
-
-    type: boolean
-
-  - Response not needed
-
-- #### V1.FLOWS.ON_CONTINUE
-  Will be sent from the shell host to flow app when user continue to the next flow step. Flow app should save(if needed) it's data, prepare output variables and send it in response to this event.
-
-  - Request payload
-
-    type: void
-
-  - Response payload
-
-    type: object
-    ```typescript
-    {
-      output: [
-        {
-          name: string;
-          value: any;
-        }
-      ]
-    }
-    ```
-
-  ### Generic events
+### PLUGIN SPECIFIC API
+
+ShellSDK provide a set of features which are specifically designed to allow communications with plugins running inside an application as part of the extention feature.
+
+  - #### VIEW STATE : an all instance synced data object
+
+	You might need to share between you applicatin and plugins a general context to provide coherent UI. ShellSDK let you share any `{ key: value }` object through the ViewState entity. You can define a key from any application or any plugin using the `setViewState` method. ViewState is not persistant and will be deleted when user navigate outside of the application.
+	
+	``` typescript
+	this.sdk.setViewState('TECHNICIAN', id);
+	``` 
+	
+	To listen on modification event, use the listenner `onViewState` with the expected key.
+	
+	```typescript
+	this.sdk.onViewState('TECHNICIAN', id => {
+	    this.selectedId = id;
+	}))
+	```
+	
+	A full instance of the ViewState is stored in the shell and provided on the `REQUEST_CONTEXT` event for initialisation.
+	
+	To initialise your ViewState, make sure all `.onViewState` listenners are initialise when first emitting the `REQUEST_CONTEXT` event. ShellSDK will first trigger `.on(SHELL_EVENTS.Version1. REQUEST_CONTEXT` to initialize the general context, then individually receive events on `onViewState` lsitenners. When all listenners are triggers, you will receive a `REQUEST_CONTEXT _DONE` event which could be used to eventually release the UI. 
+	
+  - #### V1.TO_APP event

```

Comment:
I don't remember we were planning such event. Do we really need it ?

Reply:
Even has been removed 😉, was a way to solve glitch on init but better to be handled by outlet with a nice loading animation.